php/libraries/NDB_BVL_Battery.class.inc

```diff
@@ -292,67 +285,49 @@ class NDB_BVL_Battery
      * Gets an associative array of instruments which are members of the current
      * battery
      *
-     * @param string  $stage        Either 'visit' or 'screening' - determines
-     *                              whether to register only screening instruments
-     *                              or visit instruments
-     * @param integer $SubprojectID The SubprojectID of that we want the battery for.
-     *
      * @return array an associative array containing Test_name,
      *         Full_name, Sub_group, CommentID
      */
-    function getBatteryVerbose($stage=null, $SubprojectID=null)
+    function getBatteryVerbose()
     {
         $DB = Database::singleton();
         // assert that a battery has already been selected
         $this->_assertBatterySelected();
 
         // craft the select query
-        $query = "SELECT f.Test_name,
-                    t.Full_name,
-                    f.CommentID,
-                    CONCAT('DDE_', f.CommentID) AS DDECommentID,
-                    ts.Subgroup_name as Sub_group,
-                    ts.group_order as Subgroup_order,
-                    t.isDirectEntry";
-        if (!is_null($stage)) {
-            $query .= ", b.instr_order as instrument_order";
-        }
-        $query  .= " FROM flag f
-            	JOIN test_names t ON (f.Test_name=t.Test_name)
-            	JOIN test_subgroups ts ON (ts.ID = t.Sub_group) ";
-        $qparams = ['SID' => $this->sessionID];
-
-        if (!is_null($stage)) {
-            $query .= "
-            	LEFT JOIN test_battery b ON (t.Test_name=b.Test_name) ";
-        }
-        $query .= " WHERE f.SessionID=:SID
-            AND f.CommentID NOT LIKE 'DDE%'";
-        if (!is_null($stage) && !is_null($SubprojectID)) {
-                $query .= " AND (
-                    b.SubprojectID IS NULL
-                    OR b.SubprojectID=:SubprojID
-                )";
-                $qparams['SubprojID'] = $SubprojectID;
-        }
-
-        if (Utility::columnsHasNull('test_subgroups', 'group_order')) {
+        $query = "SELECT DISTINCT f.Test_name,
```

Comment:
This should be converted to a group BY to avoid start next stage duplication issues

Reply:
does this still need to be resolved?

Reply:
yes but its unrelated to the survey_accounts bug if thats what you were thinking